content/tracing/languages/java.md

```diff
@@ -0,0 +1,271 @@
+
+If you aren't using a [supported framework instrumentation](#integrations), or you would like additional depth in your application’s traces, you may want to to manually instrument your code.  
+Do this either using the [Trace annotation](#trace-annotation) for simple method call tracing or with the [OpenTracing API](#opentracing-api) for complex tracing.
+
+### Trace Annotation
+
+Add the `dd-trace-api` dependency to your project. For Maven, add this to `pom.xml`:
+
+```xml
+<dependency>
+    <groupId>com.datadoghq</groupId>
+    <artifactId>dd-trace-api</artifactId>
+    <version>{version}</version>
+</dependency>
+```
+
+For Gradle, add:
+
+```gradle
+compile group: 'com.datadoghq', name: 'dd-trace-api', version: {version}
+```
+
+Now add `@Trace` to methods to have them be traced when running with `dd-java-agent.jar`.  If the agent is not attached, this annotation will have no effect on your application.
+
+### OpenTracing API
+
+Use the [OpenTracing API](https://github.com/opentracing/opentracing-java) and the Datadog Tracer (dd-trace-ot) library to measure execution times for specific pieces of code. This lets you trace your application more precisely than you can with the Java Agent alone.
+
+#### Setup
+
+For Maven, add this to `pom.xml`:
+
+```xml
+<!-- OpenTracing API -->
+<dependency>
+    <groupId>io.opentracing</groupId>
+    <artifactId>opentracing-api</artifactId>
+    <version>0.31.0</version>
+</dependency>
+
+<!-- OpenTracing Util -->
+<dependency>
+    <groupId>io.opentracing</groupId>
+    <artifactId>opentracing-util</artifactId>
+    <version>0.31.0</version>
+</dependency>
+
+<!-- Datadog Tracer (only needed if you do not use dd-java-agent) -->
+<dependency>
+    <groupId>com.datadoghq</groupId>
+    <artifactId>dd-trace-ot</artifactId>
+    <version>${dd-trace-java.version}</version>
+</dependency>
+```
+
+For Gradle, add:
+
+```
+compile group: 'io.opentracing', name: 'opentracing-api', version: "0.31.0"
+compile group: 'io.opentracing', name: 'opentracing-util', version: "0.31.0"
+compile group: 'com.datadoghq', name: 'dd-trace-ot', version: "${dd-trace-java.version}"
+```
+
+Configure your application using environment variables or system properties as discussed in the [configuration](#configuration) section.
+
+#### Custom Instrumentation Examples 
+
+Use a combination of these if the automatic instrumentation isn’t providing you enough depth or detail.
+
+Using try-finally:
+
+```java
+import io.opentracing.util.GlobalTracer;
+
+class InstrumentedClass {
+
+    void method0() {
+        /* 
+        1. Configure your application using environment variables or system properties
+        2. Using dd-java-agent (-javaagent:/path/to/dd-java-agent.jar),
+        GlobalTracer is automatically instantiated.
+        */
+        Tracer tracer = GlobalTracer.get();
+
+        Scope scope = tracer.buildSpan("operation-name").startActive(true);
+        try {
```

Comment:
Could you please indent the code in the `try`/`finally` an extra two spaces?

```java
import io.opentracing.util.GlobalTracer;

class InstrumentedClass {

    void method0() {
        /* 
         * 1. Configure your application using environment variables or system properties
         * 2. Using dd-java-agent (-javaagent:/path/to/dd-java-agent.jar),
         *    GlobalTracer is automatically instantiated.
         */
        Tracer tracer = GlobalTracer.get();

        Scope scope = tracer.buildSpan("operation-name").startActive(true);
        try {
            scope.span().setTag(DDTags.SERVICE_NAME, "my-new-service");

            // The code you're tracing
            Thread.sleep(1000);

        // If you don't call close(), the span data will NOT make it to Datadog!
        finally {
            scope.close();
        }
    }
}
```

Reply:
👍